app/components/Search/mazemapAutocomplete.js

```diff
@@ -0,0 +1,113 @@
+// @flow
+
+import React, { Component, type ComponentType } from 'react';
+import { debounce } from 'lodash';
+import { stripHtmlTags } from './utils.js';
+import "node_modules/mazemap/mazemap.min.css";
+import * as Mazemap from 'mazemap';
+
+type InjectedProps = {
+  mazemapSearch: (query: string) => Promise<*>,
+  meta: any,
+};
+
+type State = {
+  searching: boolean,
+  result: Array<Object>,
+};
+
+function mazemapAutocomplete<Props>({
+  WrappedComponent,
+}: {
+  WrappedComponent: ComponentType<Props>,
+}) {
+  const displayName =
+    WrappedComponent.displayName || WrappedComponent.name || 'Unknown';
+
+  const mySearch = new Mazemap.Search.SearchController({
```

Comment:
Maybe call this something else 😄 Something descriptive that fits with the params or to indicate that it's a mazemap search